cpp/log/frontend_test.cc

```diff
@@ -162,8 +163,11 @@ TYPED_TEST(FrontendTest, TestSubmitValid) {
   CertChain chain(this->leaf_pem_);
   EXPECT_TRUE(chain.IsLoaded());
 
+  LogEntry entry;
   SignedCertificateTimestamp sct;
-  EXPECT_OK(this->frontend_.QueueX509Entry(&chain, &sct));
+  EXPECT_OK(this->frontend_.QueueProcessedEntry(
+      this->submission_handler_.ProcessX509Submission(&chain, &entry), entry,
```

Comment:
Passing the same `entry` twice, with one of them looking "by value" (yeah, I know, thank you C++!), that looked rather alarming.

To my eye, it really looks like `ProcessX509Submission` and `ProcessPreCertSubmission` want to return a `StatusOr<LogEntry>`...